dashboard/app/api_test.go

```diff
@@ -202,3 +208,98 @@ func TestEmergentlyStoppedCrashReport(t *testing.T) {
 	c.expectOK(err)
 	c.expectEQ(len(listResp.List), 0)
 }
+
+func TestUpdateReportingPriority(t *testing.T) {
+	bug := &Bug{
+		Namespace: testConfig.DefaultNamespace,
+		Title:     "bug",
+	}
+	build := Build{
+		Namespace:    testConfig.DefaultNamespace,
+		KernelRepo:   "git://syzkaller.org",
+		KernelBranch: "branch10",
+	}
+
+	crashes := []*Crash{
+		// Revoked and no repro.
+		{
+			BuildID:        "0",
+			ReproIsRevoked: true,
+		},
+		// Non-revoked and no repro. Should have the same priority as "0".
+		{
+			BuildID: "1",
+		},
+		// Non-revoked and no repro but title matches with bug.
+		{
+			BuildID: "2",
+			Title:   bug.Title,
+		},
+		// Revoked but has syz repro.
+		{
+			BuildID:        "3",
+			ReproIsRevoked: true,
+			ReproSyz:       1,
+		},
+		// Revoked but has C repro.
+		{
+			BuildID:        "4",
+			ReproIsRevoked: true,
+			ReproC:         1,
+		},
+		// Revoked and has C repro and title matches with bug.
+		{
+			BuildID:        "5",
+			ReproC:         1,
+			Title:          bug.Title,
+			ReproIsRevoked: true,
+		},
+		// Non-revoked and has syz repro.
+		{
+			BuildID:  "6",
+			ReproSyz: 1,
+		},
+		// Non-revoked and has C repro.
+		{
+			BuildID: "7",
+			ReproC:  1,
+		},
+		// Non-revoked and has C repro and title matches with bug.
+		{
+			BuildID: "8",
+			ReproC:  1,
+			Title:   bug.Title,
+		},
+		// Last. Non-revoked, has C repro, title matches with bug and arch is AMD64.
+		{
+			BuildID: "9",
+			ReproC:  1,
+			Title:   bug.Title,
+		},
+	}
+
+	ctx := context.Background()
+	for i, crash := range crashes {
+		crash.Manager = "special-obsoleting"
+		if i == len(crashes)-1 {
+			build.Arch = targets.AMD64
+		}
+		crash.UpdateReportingPriority(ctx, &build, bug)
+	}
+
+	sort.Slice(crashes, func(i, j int) bool {
```

Comment:
Nit: if the original slice is already sorted in the order of the expected priority, you can use https://pkg.go.dev/sort#SliceIsSorted to make code simpler